File: server/api/task.go
package api

import "gopkg.in/guregu/null.v4"

type Task struct {
	Id           uint      `json:"id" form:"id"`
	Name         string    `json:"name" form:"name"`
	Status       int       `json:"status" form:"status"`
	Owner        uint      `json:"owner" form:"owner"`
	PStartAt     null.Time `json:"pStartAt" form:"pStartAt"`
	PEndAt       null.Time `json:"pEndAt" form:"pEndAt"`
	StartAt      null.Time `json:"startAt" form:"startAt"`
	EndAt        null.Time `json:"endAt" form:"endAt"`
	ProjectId    uint      `json:"projectId" form:"projectId"`
	TaskGroupId  uint      `json:"taskGroupId" form:"taskGroupId"`
	Serial       int       `json:"serial" form:"serial"`
	Desc         string    `json:"desc" form:"desc"`
	CommentCount int64     `json:"commentCount" form:"commentCount"`
}

type MoveTaskReq struct {
	Id          uint `json:"id" form:"id"`
	Prev        uint `json:"prev" form:"prev"`
	Next        uint `json:"next" form:"next"`
	TaskGroupId uint `json:"taskGroupId" form:"taskGroupId"`
}

type MoveTaskResp struct {
}

File: web/src/api/task.ts
import http from './axios'

export interface Task {
  id?: number
  projectId?: number
  taskGroupId?: number
  name?: string
  serial?: number
  status?: number
  startAt?: string
  endAt?: string
  desc?: string
}

const url = '/api/task'
export const addTask = (t: Task) => {
  return http({
    method: 'POST',
    url: url,
    data: t,
  }).then((res) => res.data)
}

export const updateTaskStatus = (t: Task) => {
  return http({
    method: 'PUT',
    url: url + '/status',
    data: t,
  }).then((res) => res.data)
}

export const moveTask = (opt: {
  id: number
  prev: number
  next: number
  taskGroupId: number
}) => {
  return http({
    method: 'POST',
    url: url + '/move',
    data: opt,
  }).then((res) => res.data)
}

export const updateTask = (t: Task) => {
  return http({
    method: 'PUT',
    url: url,
    data: t,
  }).then((res) => res.data)
}

export const delTask = (t: Task) => {
  return http({
    method: 'DELETE',
    url: url,
    data: t,
  }).then((res) => res.data)
}
